plugins/clock/CMakeLists.txt

```diff
@@ -0,0 +1,15 @@
+project(clock)
+
+add_library(gz-sim-yarp-clock-system SHARED Clock.cc)
+target_compile_features(gz-sim-yarp-clock-system PRIVATE cxx_std_17)
+
+target_link_libraries(gz-sim-yarp-clock-system
+  PRIVATE gz-plugin${GZ_PLUGIN_VER}::gz-plugin${GZ_PLUGIN_VER}
+  PRIVATE gz-sim${GZ_SIM_VER}::gz-sim${GZ_SIM_VER}
+  PRIVATE
+    ${YARP_LIBRARIES}
```

Comment:
```suggestion
```

If you need more `YARP` targets, let's list them explicitly (probably `YARP::YARP_dev` and/or `YARP::YARP_sig` are necessary).

Reply:
@traversaro could you point me towards some reference in which to better understand this? I've seen in Yarp documentation the usage of `${YARP_LIBRARIES}`: https://yarp.it/latest/group__add__a__plugin.html

Reply:
Good point, in general this would need to be updated. The point is that you never actually know what the `${YARP_LIBRARIES}` variable will contain, as it depends on how `find_package(YARP)` has been called (with which arguments). So in so-called "Modern CMake" it is better to always use targets, that clearly identify a single library. In your case, just try to put `YARP::YARP_dev` in place of `${YARP_LIBRARIES}`, and if there are compilation error we can look into that.

Reply:
Solve in 8f6967faa26c75debd5b0443e3d0180ca9f9d55e

Putting YARP::YARP_dev in place of ${YARP_LIBRARIES} worked.